packages/run-protocol/test/amm/vpool-xyk-amm/test-amm-calc.js

```diff
@@ -33,6 +47,13 @@ test('balancesToReachRatio calculations are to spec', t => {
         giveY: arbGiveY,
       }),
       ({ poolX, poolY, giveX, giveY }) => {
+        const kBefore = Number(multiply(poolX.value, poolY.value));
+        const poolRatioAfter = Math.max(
+          Number(giveX.value + poolX.value) / Number(giveY.value + poolY.value),
+          Number(giveY.value + poolY.value) / Number(giveX.value + poolX.value),
+        );
+        // skip cases where the target ratio is more than K at the start
+        fc.pre(poolRatioAfter < kBefore);
```

Comment:
does this `fc.pre` check subsume the `imbalancedRequest(` check below?

Reply:
It's exactly the same calculation.  thanks.